src/wallet/rpcwallet.cpp

```diff
@@ -4295,13 +4425,17 @@ UniValue z_getbalanceforaccount(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
-    if (fHelp || params.size() < 1 || params.size() > 2)
+    if (fHelp || params.size() < 1 || params.size() > 3)
         throw runtime_error(
-            "z_getbalanceforaccount account ( minconf )\n"
+            "z_getbalanceforaccount account ( minconf asOfHeight )\n"
             "\nReturns the account's spendable balance for each value pool (\"transparent\", \"sapling\", and \"orchard\")."
             "\nArguments:\n"
             "1. account      (numeric) The account number.\n"
             "2. minconf      (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "3. asOfHeight       (numeric, optional) Execute the query as if it were run when\n"
+            "                    the blockchain was at the height specified by this argument.\n"
+            "                    The default is to use the entire blockchain that the node is\n"
+            "                    aware of.\n"
```

Comment:
```suggestion
            "3. asOfHeight   (numeric, optional) Execute the query as if it were run when\n"
            "                the blockchain was at the height specified by this argument.\n"
            "                The default is to use the entire blockchain that the node is\n"
            "                aware of.\n"
```
Document what happens if `asOfHeight` is greater than the height of the last known block.

Reply:
Formatting change no longer relevant.